ValveResourceFormat/Resource/ResourceTypes/KV3.cs

```diff
@@ -0,0 +1,590 @@
+            }
+            //String opening
+            else if (c == '"')
+            {
+                stateStack.Pop();
+                stateStack.Push(State.VALUE_STRING);
+                currentString.Clear();
+            }
+            //Booleans
+            else if ((c == 'f' && ( fileStream[index+1] == 'a' && fileStream[index+2] == 'l' && fileStream[index+3] == 's' && fileStream[index+4] == 'e' )) 
+                || (c == 't' && (fileStream[index+1] == 'r' && fileStream[index+2] == 'u' && fileStream[index+3] == 'e')))
+            {
+                stateStack.Pop();
+                stateStack.Push(State.VALUE_BOOL);
+                currentString.Clear();
+                currentString.Append(c);
+            }
+            //Number
+            else if (numerals.Contains(c))
+            {
+                stateStack.Pop();
+                stateStack.Push(State.VALUE_NUMBER);
+                currentString.Clear();
+                currentString.Append(c);
+            }
+            //Flagged resource
+            else
+            {
+                stateStack.Pop();
+                stateStack.Push(State.VALUE_FLAGGED);
+                currentString.Clear();
+                currentString.Append(c);
+            }
+        }
+
+        //Reading a property name
+        private static void ReadPropName(char c)
+        {
+            //Stop once whitespace is encountered
+            if (whiteSpace.Contains(c))
+            {
+                stateStack.Pop();
+                stateStack.Push(State.SEEK_VALUE);
+                currentName = currentString.ToString();
+                return;
+            }
+
+            currentString.Append(c);
+        }
+
+        //Read a structure
+        private static void ReadValueStruct(char c)
+        {
+            //Ignore whitespace
+            if (whiteSpace.Contains(c))
+            {
+                return;
+            }
+
+            //Catch comments
+            if (c == '/')
+            {
+                stateStack.Push(State.COMMENT);
+                currentString.Clear();
+                currentString.Append(c);
+                return;
+            }
+
+            //Check for the end of the structure
+            if (c == '}')
+            {
+                KVObject value = objStack.Pop();
+                objStack.Peek().AddProperty(value.key, new KVValue(typeof(KVObject), value));
+                stateStack.Pop();
+                return;
+            }
+
+            //Start looking for the next property name
+            stateStack.Push(State.PROP_NAME);
+            currentString.Clear();
+            currentString.Append(c);
+        }
+        
+        //Read a string value
+        private static void ReadValueString(char c)
+        {
+            if (c == '"')
+            {
+                //Multiline string detected
+                if (currentString.ToString() == "\"")
```

Comment:
Wouldn't this treat the following as the start of a multistring and get confused
```
{
    emptyString = ""
    somethingElse = true
}
```

Note, Valves documentation states its only a start of a Multistring when [it starts with three unescaped quotes followed by an immediate newline (eg. the buffer looks like `\"\"\"\n`)](https://developer.valvesoftware.com/wiki/Dota_2_Workshop_Tools/KeyValues3#Multiline_String_Literals)